pkg/actions/split/do.go

```diff
@@ -0,0 +1,57 @@
+package split
+
+import (
+	"context"
+	dstk "github.com/anujga/dstk/pkg/api/proto"
+	"github.com/anujga/dstk/pkg/core"
+	"github.com/anujga/dstk/pkg/ss/partition"
+)
+
+type Dag struct {
+	Part *dstk.Partition
+	IdGenerator func() int64
+	SeRpc dstk.PartitionRpcClient
+}
+
+func (d *Dag) Start(ctx context.Context, splitPoint []byte) *core.FutureErr {
+	p := core.NewPromise()
+	return p.Complete(func() error {
+		partId := d.Part.GetId()
+		split1Id := d.IdGenerator()
+		d.createFollower(split1Id, partId, d.Part.GetStart(), splitPoint)
+
+		split2Id := d.IdGenerator()
+		d.createFollower(split2Id, partId, splitPoint, d.Part.GetEnd())
+
+		d.checkState(split1Id, partition.Follower).Wait()
+		d.checkState(split2Id, partition.Follower).Wait()
+
+		d.makeProxy(partId, []int64{split1Id, split2Id})
+		d.checkState(partId, partition.Proxy).Wait()
```

Comment:
wait should accept context since its a blocking call. its okay not to respect it for now